### ISSUE TITLE
version 1.9.5-r1 release does not view because the url is https://www.example.com/

### ISSUE THREAD

**Issue by rene-gomez** [bitnami/bitnami-docker-discourse#41]
**Description**

the images sometimes works, because  the image url used is a relative url (/images/....), which works, but after a page reload the image url changes to hardcode www.example.com as the host (//www.example.com/images/) instead of the actual domain we are using.

For example, this is how an emoji is displaying:

"img src="//www.example.com/images/emoji/emoji_one/open_mouth.png "

I edited /bitnami/discourse/conf/database.yml and discourse.conf on discourse and sidekiq, but does not work.

**Version**

- Output of `docker version`:

```
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

- Output of `docker info`:

```
Containers: 26
 Running: 26
 Paused: 0
 Stopped: 0
Images: 10
Server Version: 17.05.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 154
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9048e5e50717ea4497b757314bad98ea3763c145
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-96-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 5.827GiB
Name: docker
ID: 4B6I:L3ET:6WWL:MFHG:R6Q7:O7FL:4YYV:233Y:FJQY:M3U7:UR5Y:QAZA
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
```

- Output of `docker-compose version` (if applicable):

```
Containers: 26
 Running: 26
 Paused: 0
 Stopped: 0
Images: 10
Server Version: 17.05.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 154
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9048e5e50717ea4497b757314bad98ea3763c145
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-96-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 5.827GiB
Name: docker
ID: 4B6I:L3ET:6WWL:MFHG:R6Q7:O7FL:4YYV:233Y:FJQY:M3U7:UR5Y:QAZA
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
```

**Comment by javsalgar** [comment]
Hi, 

Did you try executing rake db:migrate inside the container?

**Comment by rene-gomez** [comment]
Hi @javsalgar can you show me the steps for this change?, what container for example?

best regards.

**Comment by javsalgar** [comment]
It should be executed in the discourse container (the one that exposes port 80->3000 in docker ps). Execute the following:

```
cd /opt/bitnami/discourse
bin/bundle exec rake db:migrate RAILS_ENV=production
```

**Comment by rene-gomez** [comment]
@javsalgar this is the result:

`/usr/bin/env: ruby: No such file or directory`

**Comment by javsalgar** [comment]
Hi,

This is weird. I was able to run the commands without issues. I tried your version of the container and ruby is indeed there.

```
Status: Downloaded newer image for bitnami/discourse:1.9.5-r1



Welcome to the Bitnami discourse container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-discourse
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-discourse/issues
Send us your feedback at containers@bitnami.com

root@f321f50e0c0d:/# 
root@f321f50e0c0d:/# 
root@f321f50e0c0d:/# env ruby
^Cruby: Interrupt

root@f321f50e0c0d:/# which ruby
/opt/bitnami/ruby/bin/ruby
root@f321f50e0c0d:/# cd /opt/bitnami/discourse
root@f321f50e0c0d:/opt/bitnami/discourse# bin/bundle exec rake db:migrate RAILS_ENV=production
```

**Comment by javsalgar** [comment]
This is weird, I was able to reproduce the issue and tried migrating the database, changing the defaults, exploring the database, restarting sidekiq... But still the same issue appears. This will have to be looked into in more detail. I am sorry but I cannot give an ETA, but this issue will be considered in future releases.

**Comment by rene-gomez** [comment]
Hi @javsalgar  please label this issue that a bug, because is it

Best Regards.

**Comment by rene-gomez** [comment]
Hi @javsalgar , ¿do you have any progress?, in the roadmap this issue is fixet yet?

¿do you have  date for the image fixed is release?

best regards.

**Comment by carrodher** [comment]
Hi,
We have at task included in our backlog. As the impact of the issue is not critical, there are no immediate plans to work on it. Nevertheless, it will be considered for future application versions.

We will update this thread as soon as we have more news.

**Comment by rene-gomez** [comment]
hi @carrodher the issue persis on 2.0.3 version, when the system send the email activation for new user the url is www.example.com, **of course of the issue is critical, because new users cannot activate their account.**

do you have news?

best regards

**Comment by beltran-rubo** [comment]
I posted a workaround for configuring the domain at https://github.com/bitnami/bitnami-docker-discourse/issues/47#issuecomment-410776027

**Comment by beltran-rubo** [comment]
Hi @rene-gomez 

We released a new revision "2.0.3-r11" where the domain in the email notifications is configurable using the ENV var `DISCOURSE_HOSTNAME`. The default value is `www.example.com`.
